### DISCUSSION TITLE
Global Variables

### DISCUSSION THREAD

**Discussion by sistematico** [nuxt/nuxt#19896] [Questions] [answered]
Is there anyway to use global variable/object/function without:

`const someName = useSomething()`

On every component/page?

Thank you very much.

**Comment by TheAlexLichter** [comment, marked as answer]
Can you give more info on what you try to achieve? 

You can e.g. use `provide`/`inject` via Nuxt plugin, or use a composable with useState that will share state.

In your `composables/useColor.ts` file:

`export const useColor = () => useState('color', 'rebeccapurple')`

Somewhere in code:
`useColor().value = 'hotpink`

Wherever `useColor()` is used, the value will be changed.

See also [more in the docs](https://nuxt.com/docs/getting-started/state-management#shared-state)

**Reply by sistematico** [reply]
@manniL thank you very much for your response.

I'm trying to achieve this: https://nuxt.com/docs/getting-started/configuration#app-configuration

Without defining: `const appConfig = useAppConfig()` on every component.

Example: 
```
<template>
<h1>App Name: {{ appConfig.appName }}</h1>
</template>
```

or 

```
<template>
<h1>Theme Color: {{ appConfig.theme }}</h1>
</template>
```

**Reply by TheAlexLichter** [reply]
For now you have to define it in every component but please feel free to raise a feature request to have a template shortcut for it.
